aptos-move/framework/src/natives/aggregator_natives/aggregator_v2.rs

```diff
@@ -443,6 +443,15 @@ fn native_create_snapshot(
     let snapshot_type = SnapshotType::from_ty_arg(context, &ty_args[0])?;
     let input = snapshot_type.pop_snapshot_value_by_type(&mut args)?;
 
+    if let SnapshotValue::String(v) = &input {
+        if v.len() > STRING_SNAPSHOT_INPUT_MAX_LENGTH {
+            return Err(SafeNativeError::Abort {
+                abort_code: EINPUT_STRING_LENGTH_TOO_LARGE,
+            });
+        }
+        context.charge(AGGREGATOR_V2_CREATE_SNAPSHOT_PER_BYTE * NumBytes::new(v.len() as u64))?;
```

Comment:
We probably should charge before we return an error.

Reply:
in that case, should the limit be charged and not v.len()?

Reply:
everywhere we first get the bytes from input, and then charge the bytes. so we should charge the correct len.

I assume maybe we somewhere charge the input size itself.